src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemEventArgs.cs

```diff
@@ -17,14 +17,18 @@ public class FileSystemEventArgs : EventArgs
         /// </devdoc>
         public FileSystemEventArgs(WatcherChangeTypes changeType, string directory, string? name)
         {
+            ArgumentNullException.ThrowIfNull(directory);
+
             _changeType = changeType;
             _name = name;
-            _fullPath = Path.Join(Path.GetFullPath(directory), name);
 
-            if (string.IsNullOrWhiteSpace(name))
+            _fullPath = Path.GetFullPath((string.IsNullOrWhiteSpace(directory), string.IsNullOrWhiteSpace(name)) switch
```

Comment:
Consider not using the switch, keep the code as it was before, and just use "." when `directory` is empty (or when whitespace too).

```cs
if (string.IsNullOrWhiteSpace(directory)) // empty directory means current working dir.
{
    directory = ".";
}

_fullPath = Path.Join(Path.GetFullPath(directory), name);

if (string.IsNullOrWhiteSpace(name))
{
    _fullPath = PathInternal.EnsureTrailingSeparator(_fullPath);
}
```